### ISSUE TITLE
Is 'mdx' using the right grammar?

### ISSUE THREAD

**Issue by pomber** [shikijs/shiki#237]
## VS Code (with light-plus theme) shows:
<img src="https://user-images.githubusercontent.com/1911623/135264534-6d138eb3-5d2d-4977-8239-cdd9fb7bf026.png" width="600"/>

## Shiki shows:
<img src="https://user-images.githubusercontent.com/1911623/135264614-22b5bce0-83f1-40e6-9645-0ced2881c2ad.png" width="600"/>

---

My highlighting code just in case:

```js
const shiki = await import("shiki")
const highlighter = await shiki.getHighlighter({  theme: "light-plus" })
const html = highlighter.codeToHtml(mdxSource, "mdx"),
```

AFAIK the mdx extension I'm using in vscode uses the [same grammar as shiki](https://github.com/shikijs/shiki/blob/2c4c5c682152ed3ee9bd30bedbbc9f8f64ad3ae1/scripts/grammarSources.ts#L127).

**Comment by muenzpraeger** [comment]
@pomber I know this is an old issue - is it still valid for the most current version of Shiki?

**Comment by pomber** [comment]
> @pomber I know this is an old issue - is it still valid for the most current version of Shiki?

Haven't tried again, but it probably is still an issue. The [mdx grammar](https://github.com/shikijs/shiki/blob/main/scripts/grammarSources.ts#L187) that shiki is using is from [a deprecated project](https://github.com/silvenon/vscode-mdx/). 

There's a [new grammar](https://github.com/mdx-js/vscode-mdx/tree/main/packages/vscode-mdx/syntaxes) being developed by @remcohaszing but I'm not sure if it makes sense to switch it.

**Comment by muenzpraeger** [comment]
Thanks for the info. Just checked with the new grammar, and yes, same behavior.

Generally I think it's good to switch, as it's the new standard grammar (and referencing deprecated ones isn't... great).

I'll dive a bit more into why the MDX ref doesn't work.

**Comment by muenzpraeger** [comment]
After some digging the root cause is that the tokenization does not return the expected tokens with the grammar. Which is not a grammar issue, IMHO.

I can get it to work when using:
- 4 instead of 3 backticks
- 3 tildes instead of backticks

So somewhere along the way there is an issue with the backticks. Will dive more into that.

**Comment by remcohaszing** [comment]
I want to clarify I'm **not** developing a new MDX grammar. I'm working on that repo, but don't want to claim credit for the grammar. I just moved it within the `vscode-mdx` repository because I did some internal restructuring. But yes, the one by @silvenon has been deprecated in favour of that one.

I do believe it's good to use that grammar as the source of truth. It has some bugs though, but I'm not very familiar with tmLanguage, so I rely on external contributors to solve them (for now).

**Comment by muenzpraeger** [comment]
Yeah, the issue is a level before.

The MDX grammar references the standard Markdown, which is what detects code fences.

**Comment by LeoDog896** [comment]
This has been fixed by #452.

**Comment by octref** [comment]
Closing as fixed by #452.
